lib/WorldState.js

```diff
@@ -1,44 +1,46 @@
-const Mission = require('./Mission');
-const Reward = require('./Reward');
-const VoidTrader = require('./VoidTrader');
-const DailyDeal = require('./DailyDeal');
-const Simaris = require('./Simaris');
-const ConclaveChallenge = require('./ConclaveChallenge');
-const PersistentEnemy = require('./PersistentEnemy');
-const timeDate = require('./timeDate');
-const translator = require('./translation');
-const EarthCycle = require('./EarthCycle');
-const CetusCycle = require('./CetusCycle');
-const ConstructionProgress = require('./ConstructionProgress');
-const VallisCycle = require('./VallisCycle');
-const WeeklyChallenge = require('./WeeklyChallenge');
-const Nightwave = require('./Nightwave');
-const Kuva = require('./Kuva');
-const SentientOutpost = require('./SentientOutpost');
-const CambionCycle = require('./CambionCycle');
-const SteelPathOffering = require('./SteelPathOffering');
-
-/* eslint-disable no-unused-vars */
+import { createRequire } from 'module';
+import Alert from './Alert.js';
+import CambionCycle from './CambionCycle.js';
+import CetusCycle from './CetusCycle.js';
+import ConclaveChallenge from './ConclaveChallenge.js';
+import ConstructionProgress from './ConstructionProgress.js';
+import DailyDeal from './DailyDeal.js';
+import DarkSector from './DarkSector.js';
+import DarkSectorBattle from './DarkSectorBattle.js';
+import EarthCycle from './EarthCycle.js';
+import Fissure from './Fissure.js';
+import FlashSale from './FlashSale.js';
+import GlobalUpgrade from './GlobalUpgrade.js';
+import Invasion from './Invasion.js';
+import Kuva from './Kuva.js';
+import Mission from './Mission.js';
+import News from './News.js';
+import Nightwave from './Nightwave.js';
+import PersistentEnemy from './PersistentEnemy.js';
+import Reward from './Reward.js';
+import SentientOutpost from './SentientOutpost.js';
+import Simaris from './Simaris.js';
+import Sortie from './Sortie.js';
+import SortieVariant from './SortieVariant.js';
+import SteelPathOffering from './SteelPathOffering.js';
+import SyndicateMission from './SyndicateMission.js';
+import VallisCycle from './VallisCycle.js';
+import VoidTrader from './VoidTrader.js';
+import WeeklyChallenge from './WeeklyChallenge.js';
+import WorldEvent from './WorldEvent.js';
+
+// Utils
+import * as timeDate from './timeDate.js';
+import * as translator from './translation.js';
 // needed for type declarations
-const Dependency = require('./supporting/Dependency');
-const ExternalMission = require('./supporting/ExternalMission');
-const MarkdownSettings = require('./supporting/MarkdownSettings');
+/* eslint-disable no-unused-vars */
+import Dependency from './supporting/Dependency.js';
+import ExternalMission from './supporting/ExternalMission.js';
+import MarkdownSettings from './supporting/MarkdownSettings.js';
+
+// JSON Data
+const req = createRequire(import.meta.url);
+const sortieData = req('warframe-worldstate-data/data/sortieData.json');
```

Comment:
yeaaaah, i was afraid of that... i don't necessarily wanna move to this until we can import with json assertion

Reply:
It definitely felt weird implementing this, but we could potentially achieve it with a simple JSON npm package?

Reply:
yes, but also, i'd rather wait for native support, because this is going to be _super_ problematic on others, like warframe-worldstate-data, where all i'm doing is exporting json (because this is kinda the test project for it)

Reply:
It seems that they are adding support for import JSON assertions in [v17.1.0](https://nodejs.org/api/esm.html#import-assertions).

Example:
```Javascript
import fooData from './foo.json' assert { type: 'json' };

const { default: barData } =
  await import('./bar.json', { assert: { type: 'json' } });
```
> From Node.js [Documentation](https://nodejs.org/api/esm.html#import-assertions)

Hopefully they can get a stable release of v17 out soon sometime in the next year or so.